src/coreclr/jit/assertionprop.cpp

```diff
@@ -4145,20 +4146,29 @@ GenTree* Compiler::optAssertionProp_ModDiv(ASSERT_VALARG_TP assertions, GenTreeO
         changed = true;
     }
 
-    if (op2IsNotZero)
+    if (((tree->gtFlags & GTF_DIV_MOD_NO_BY_ZERO) == 0) && op2IsNotZero)
     {
         JITDUMP("Divisor for DIV/MOD is proven to be never negative...\n")
         tree->gtFlags |= GTF_DIV_MOD_NO_BY_ZERO;
         changed = true;
     }
 
-    if (op1IsNotNegative || op2IsNotNegative)
+    if (((tree->gtFlags & GTF_DIV_MOD_NO_OVERFLOW) == 0) && (op1IsNotNegative || op2IsNotNegative))
     {
         JITDUMP("DIV/MOD is proven to never overflow...\n")
         tree->gtFlags |= GTF_DIV_MOD_NO_OVERFLOW;
         changed = true;
     }
 
+    if (((tree->gtFlags & GTF_UMOD_UINT16_OPERANDS) == 0) && tree->OperIs(GT_UMOD) && op2->IsCnsIntOrI() &&
+        op2->AsIntCon()->IconValue() <= UINT16_MAX &&
+        IntegralRange::ForNode(op1, this).GetUpperBound() <= SymbolicIntegerValue::UShortMax)
```

Comment:
check lower bound for being 0?

Reply:
Isn't that implied by `GT_UMOD`?

Reply:
I guess so? still worth leaving an assert at least I guess

Reply:
Guess not 😄
> Assertion failed 'IntegralRange::ForNode(op1, this).GetLowerBound() >= SymbolicIntegerValue::Zero'